src/compiler/config/validate-service-worker.ts

```diff
@@ -22,11 +42,18 @@ export const validateServiceWorker = (config: d.ValidatedConfig, outputTarget: d
     return;
   }
 
-  if (typeof outputTarget.serviceWorker !== 'object') {
-    // what was passed in could have been a boolean
-    // in that case let's just turn it into an empty obj so Object.assign doesn't crash
-    outputTarget.serviceWorker = {};
-  }
+  const globDirectory =
+    typeof outputTarget.serviceWorker?.globDirectory === 'string'
```

Comment:
Do we have a guarantee here that `outputTarget.serviceWorker` is an object here? I don't believe it's part of `ValidatedConfig`, and (TBH) I'm not sure if the comment in the removed `if` statement holds true or not

Reply:
we don't have such a guarantee, but I think the optional chaining here takes care of that - if `.serviceWorker` returns undefined then iirc it will short-circuit the `?.globDirectory` and the whole expression (`outputTarget.serviceWorker?.globDirectory`) will evaluate to `undefined`

Reply:
also this test will test this case, where `.serviceWorker` is `undefined` and `globDirectory` should then be set to `appDir`:

https://github.com/ionic-team/stencil/blob/cac647b67cb655477646fda072c5693c9df7f159/src/compiler/config/test/validate-service-worker.spec.ts#L81-L88